benchexec/tools/sikraken.py

```diff
@@ -0,0 +1,53 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+This class serves as BenchExec tool adaptor for Sikraken for Test-Comp
+See baseclass documentation at https://github.com/sosy-lab/benchexec/blob/main/benchexec/tools/template.py
+"""
+import os
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    BenchExec tool-info for Sikraken
+    """
+
+    def name(self):
+        return "Sikraken" 
+    
+    def project_url(self):
+        return "https://github.com/echancrure/Sikraken" 
+        
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("sikraken.sh", subdir="bin")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, "-v")  
+    
+    """
+    #Assuming that the default working directory is the root of the project (i.e. sikraken/) we don't need this
+    def working_directory(self, executable):
+        executableDir = os.path.dirname(executable)
+        return executableDir
+    """
+
+    #Sikraken is currently ignoring property files as we only support "coverage-branches.prp"
+    #I am assuming more options come from the benchmark definition .xml file "The benchmark definition defines the categories that the tester is to be executed on, the parameters to be passed, and the resource limits."
+    #the rlimits for testcomp are passed as options
```

Comment:
Not sure what you mean here. The resource limits are passed in the parameter `rlimits`, not in the parameter `options`. It is also not clear to me what the goal of this comment is, should it just be removed?

Reply:
removed the comment, what I meant was that I am ignoring rlimits as I am passing the time budget limit from the options field of benchmark definitions

Reply:
Ok, but you could also add the logic here and thus make the benchmark definitions easier?

Reply:
As I am new here, I am following the pattern of the other tool-info as much as I can

Reply:
Well, other tool-info modules for tools that accept a time limit also do this:
https://github.com/sosy-lab/benchexec/blob/7b04ac9e6f636a1f348c604a0ab772141955fd0d/benchexec/tools/cpachecker.py#L162-L163

Reply:
I see, but I am reluctant to change because:
1. I feel the less Python I write, the better given my poor knowledge and difficulty in testing.
2. Most tools don't check the limits.
3. The limit for TestComp is set at 900 seconds for all tasks.
4. it would require a change in my bench-def too.
5. I thought I 'Resolved' this conversation

If despite this, you want me to change my PR, please instruct me, and I'll try to implement the change.

Reply:
1. is no problem as the code would be simple and I provide reviews.
2. isn't really relevant, only what is best for your case needs to be considered. (And most tools do not even have a time-limit parameter, so the question does not arise for them.)
3. is correct. But if you want to make your tool-info module easier to use for other use cases of benchmarking your tool (e.g., by other researchers), it would be an argument for adding this feature.
4. is correct but trivial.
5. Resolving a discussion unilaterally is not a good workflow on GitHub. First, there are no notifications about this, so the other people involved are not notified. Second, if everyone just closes discussions they think are resolved, this will mean that it is hard for the other involved people to keep track of what was open and check whether it is really resolved. But for a reviewer this is important. Typically, a discussion should only be marked as resolved by the one who raised the concern after both sides agree.

So, I don't see any problems with adding this feature and I wanted to make sure that you are not avoiding it for reasons that can be resolved easily. But it is not a requirement to add this in order to get the PR merged.

Reply:
Thank you for your help.
I have so many other things to do on the tool side that if the current tool-info is acceptable then I am happy to leave things as is.
Thanks.